REFRAME.md

```diff
@@ -280,6 +282,45 @@ Response:
 {"PutIPNSResponse : {}"}
 ```
 
+#### GetPeerAddresses
+
+A message for getting the addresses of a libp2p peer
+
+```ipldsch
+    type GetPeerAddressesRequest struct {
+        ID Bytes # libp2p PeerID
+        RecordTypes [String]
+    }

```

Comment:
Having `RecordTypes` doesn't seem necessary as the responder could just send back all record types it knows about. Is there value in being able to ask to send back less data here? The same could be argued in other places (e.g. not sending back TransferProtocols we don't support anyway).

I'm planning to remove this, but wanted to leave this as a place for comment if there was demand for it to stay

Reply:
if we do have it, we should have some recommendation of how we identify types - what are these strings supposed to be?

It doesn't seem overly harmful to have, but it does add another point of complexity in implementation.

Reply:
+1: if we are to have it, it should be a union of sorts that becomes the formal spec of known record types.

if we are to have it, there must also me a case for "return all types", otherwise we preclude a dumb middlebox from pre-fetching data.

Reply:
on the whole, this does seem as an unnecessary micro-optimization (at complexity cost), because the number of records a peer can ever have is limited by the number of types that exist, which is a small number. this is not something that can grow arbitrarily, so ... i would vote no here. the rule of thumb being: can the number of returned results grow in a runtime-dependent manner (e.g. # of providers for a cid depends on runtime conditions, vs number of address records is always capped by the number of types of records which is a compile-time constant)